clients/wallet/src/cache.rs

```diff
@@ -0,0 +1,344 @@
+pub struct TxEnvelopeStorage {
+	path: String,
+	inner_path: String,
+}
+
+impl TxEnvelopeStorage {
+	/// returns the full directory path
+	fn full_path(&self) -> String {
+		format!("{}/{}", self.path, self.inner_path)
+	}
+}
+
+impl TxEnvelopeStorage {
+	pub fn new(path: String, public_key: &str, is_public_network: bool) -> Self {
+		let inner_path = format!("{public_key}_{is_public_network}");
+		let full_path = format!("{path}/{public_key}_{is_public_network}");
+
+		if let Err(e) = create_dir_all(&full_path) {
+			tracing::warn!("Failed to create directory of {full_path}: {:?}", e);
+		}
+		tracing::info!("path for caching stellar transactions: {full_path}");
+
+		TxEnvelopeStorage { path, inner_path }
+	}
+
+	/// Returns true if transaction was saved to a local path.
+	/// The filename will be the transaction's sequence number.
+	pub fn save_to_local(
+		&self,
+		sequence: SequenceNumber,
+		transaction: TransactionEnvelope,
+	) -> bool {
+		let full_file_path = format!("{}/{}", self.full_path(), sequence);
+
+		let path = Path::new(&full_file_path);
+		if path.exists() {
+			return false
+		}
+
+		let mut file = unwrap_or_return!(
+			OpenOptions::new().write(true).create(true).open(path),
+			false,
+			format!("Failed to create file for transaction with sequence {}", sequence)
+		);
+
+		unwrap_or_return!(
+			// writing the transaction to file in Vec<u8> format
+			write!(file, "{:?}", transaction.to_xdr()),
+			false,
+			format!("Failed to write transaction with sequence {}", sequence)
+		);
+
+		true
+	}
+
+	pub fn remove_transaction(&self, sequence: SequenceNumber) -> bool {
+		let full_file_path = format!("{}/{sequence}", self.full_path());
+		remove_file(full_file_path).is_ok()
+	}
+
+	#[doc(hidden)]
+	#[cfg(any(test, feature = "testing-utils"))]
+	/// Removes all transactions, including the directory
+	pub fn remove_all(&self) -> bool {
+		std::fs::remove_dir_all(self.full_path()).is_ok()
+	}
+
+	#[allow(dead_code)]
+	#[cfg(any(test, feature = "testing-utils"))]
+	/// Returns a transaction if a file was found, given the sequence number
+	pub fn get_tx_envelope(&self, sequence: SequenceNumber) -> Option<TransactionEnvelope> {
+		let full_file_path = format!("{}/{sequence}", self.full_path());
+		let path = Path::new(&full_file_path);
+
+		if !path.exists() {
+			return None
+		}
+		read_tx_envelope_from_path(path).map(|(tx, _)| tx)
+	}
+
+	/// Returns all transactions found in local path
+	pub fn get_tx_envelopes(&self) -> Vec<(TransactionEnvelope, SequenceNumber)> {
+		let path = self.full_path();
+		let directory =
+			unwrap_or_return!(read_dir(&path), vec![], format!("Failed to read directory {path}"));
+
+		let mut res: Vec<(TransactionEnvelope, SequenceNumber)> = directory
+			.into_iter()
+			.filter_map(|entry| {
+				let path = entry.ok()?.path();
+				extract_tx_envelope_from_path(path)
+			})
+			.collect();
+
+		// the envelopes have been filtered out with those having sequence numbers already
```

Comment:
I don't understand what this comment means. What was filtered out? I get that in the next line we order them in ascending order but I guess the comment is unrelated to that?